[Bug]: MESHTASTIC_EXCLUDE_WIFI breaks PowerFSM

### Category

WiFi

### Hardware

Heltec V3

### Is this bug report about any UI component firmware like InkHUD or Meshtatic UI (MUI)?

- [ ] Meshtastic UI aka MUI colorTFT
- [ ] InkHUD ePaper
- [ ] OLED slide UI on any display

### Firmware Version

2.6.5

### Description

When MESHTASTIC_EXCLUDE_WIFI 1 is used, PowerFSM gets unhappy.

Looks like isWifiAvailable isn't defined anymore.

### Relevant log output

```Shell
Compiling .pio/build/heltec-v3/src/PowerFSM.cpp.o
.....
src/PowerFSM.cpp: In function 'void PowerFSM_setup()':
src/PowerFSM.cpp:386:44: error: 'isWifiAvailable' was not declared in this scope
     if ((config.power.is_power_saving) && !isWifiAvailable() && !isTrackerOrSensor) {
                                            ^~~~~~~~~~~~~~~
src/PowerFSM.cpp:386:44: note: suggested alternative: 'uartAvailable'
     if ((config.power.is_power_saving) && !isWifiAvailable() && !isTrackerOrSensor) {
                                            ^~~~~~~~~~~~~~~
                                            uartAvailable
*** [.pio/build/heltec-v3/src/PowerFSM.cpp.o] Error 1
```

--- a/src/PowerFSM.cpp
+++ b/src/PowerFSM.cpp
@@ -19,7 +19,7 @@
 #include "sleep.h"
 #include "target_specific.h"
 
-#if HAS_WIFI && !defined(ARCH_PORTDUINO) || !defined(MESHTASTIC_EXCLUDE_WIFI)
+#if HAS_WIFI && !defined(ARCH_PORTDUINO) || defined(MESHTASTIC_EXCLUDE_WIFI)
 #include "mesh/wifi/WiFiAPClient.h"
 #endif
 
@@ -403,7 +403,9 @@ void PowerFSM_setup()
                                       Default::getConfiguredOrDefaultMs(config.display.screen_on_secs, default_screen_on_secs),
                                       NULL, "Screen-on timeout");
     }
-#else
+#endif // HAS_WIFI || !defined(MESHTASTIC_EXCLUDE_WIFI)
+
+#else // (not) ARCH_ESP32
     // If not ESP32, light-sleep not used. Check periodically if config has drifted out of stateDark
     powerFSM.add_timed_transition(&stateDARK, &stateDARK,
                                   Default::getConfiguredOrDefaultMs(config.display.screen_on_secs, default_screen_on_secs), NULL,
@@ -412,5 +414,4 @@ void PowerFSM_setup()
 
     powerFSM.run_machine(); // run one iteration of the state machine, so we run our on enter tasks for the initial DARK state
 }
-#endif
 #endif

--- a/src/PowerFSM.cpp
+++ b/src/PowerFSM.cpp
@@ -19,7 +19,7 @@
 #include "sleep.h"
 #include "target_specific.h"
 
-#if HAS_WIFI && !defined(ARCH_PORTDUINO)
+#if HAS_WIFI && !defined(ARCH_PORTDUINO) || !defined(MESHTASTIC_EXCLUDE_WIFI)
 #include "mesh/wifi/WiFiAPClient.h"
 #endif
 
@@ -269,9 +269,6 @@ Fsm powerFSM(&stateBOOT);
 void PowerFSM_setup()
 {
     bool isRouter = (config.device.role == meshtastic_Config_DeviceConfig_Role_ROUTER ? 1 : 0);
-    bool isTrackerOrSensor = config.device.role == meshtastic_Config_DeviceConfig_Role_TRACKER ||
-                             config.device.role == meshtastic_Config_DeviceConfig_Role_TAK_TRACKER ||
-                             config.device.role == meshtastic_Config_DeviceConfig_Role_SENSOR;
     bool hasPower = isPowered();
 
     LOG_INFO("PowerFSM init, USB power=%d", hasPower ? 1 : 0);
@@ -383,6 +380,12 @@ void PowerFSM_setup()
     // See: https://github.com/meshtastic/firmware/issues/1071
     // Don't add power saving transitions if we are a power saving tracker or sensor or have Wifi enabled. Sleep will be initiated
     // through the modules
+
+#if HAS_WIFI || !defined(MESHTASTIC_EXCLUDE_WIFI)
+    bool isTrackerOrSensor = config.device.role == meshtastic_Config_DeviceConfig_Role_TRACKER ||
+                             config.device.role == meshtastic_Config_DeviceConfig_Role_TAK_TRACKER ||
+                             config.device.role == meshtastic_Config_DeviceConfig_Role_SENSOR;
+
     if ((isRouter || config.power.is_power_saving) && !isWifiAvailable() && !isTrackerOrSensor) {
         powerFSM.add_timed_transition(&stateNB, &stateLS,
                                       Default::getConfiguredOrDefaultMs(config.power.min_wake_secs, default_min_wake_secs), NULL,
@@ -409,4 +412,5 @@ void PowerFSM_setup()
 
     powerFSM.run_machine(); // run one iteration of the state machine, so we run our on enter tasks for the initial DARK state
 }
+#endif
 #endif

